### DISCUSSION TITLE
RunSling.bat --how to edit properly for windows_server.exe?

### DISCUSSION THREAD

**Discussion by lead2crm** [GerryDazoo/Slinger#596] [Q&A] [answered]
I'm a bit confused on how to edit this properly. I am using the windows_server.exe ... where do I put the full path to slingbox_server.exe; of which, is on my C Drive? I don't see how the windows example below would ever work or even serve as an example of what to change?

```
:loop
(
REM if code is not on your C drive switch to the correct drive letter
REM g:
REM use the full pathname to where your code is located
cd slingbox
slingbox_server.exe
REM or if your using python use the full path to whatever python interpreter you're using.
REM "c:\Program Files\Python39\python.exe" slingbox_server.py 
)
goto :loop
```

**Comment by lead2crm** [comment]
Still no luck with editing this RunSLing.bat file.

Why wouldnt you just create a shortcut to slingbox_server.exe and put that in your Windows StartUp folder?

**Reply by GerryDazoo** [reply]
Sure you could do that. The batch file is there only to keep the program running if it is remotely restarted. 

but to answer your other question
change line "cd slingbox" to for examble "cd c:\the_folder\where_the_exe_is_located\slingbox_server.exe

**Reply by lead2crm** [reply]
Hmm. I thought I did that, maybe i only put the folder root on the CD line assuming that the following line below it "slingbox_server.exe" in the code example would be the reference point for the file name. I'll change this out and try again.

In regards to restarting remotely...do you mean restarting the PC remotely or restarting slingbox_server.exe remotely--If so how do we do that?

**Reply by GerryDazoo** [reply]
Sorry about that, You're right the slingbox_server.exe does NOT go on the "cd" line. 

Restarting the slingbox_server.exe, click the restart button on the web remote or use the API as documented in the release notes.

**Reply by lead2crm** [reply]
Okay, then I am definitely confused. I put

`cd C:\FolderLocation\Slinger\Slinger-main`

and it couldn't find slingbox_server.exe, which is int he "Slinger-main" folder

Example:

```
:loop
(
REM if code is not on your C drive switch to the correct drive letter
REM g:
REM use the full pathname to where your code is located
cd C:\FolderLocation\Slinger\Slinger-main
slingbox_server.exe
REM or if your using python use the full path to whatever python interpreter you're using.
REM "c:\Program Files\Python39\python.exe" slingbox_server.py 
)
goto :loop
```

**Reply by GerryDazoo** [reply]
after the cd command put in a "pwd" print working directory command to see where the batch file thinks it's at.

**Reply by lead2crm** [reply, marked as answer]
The pwd command didn't work, but it doesn't matter because I was wrong from the beginning. I thought putting RunSling.bat was necessary to restart the server on system startup, but truly it's actually to restart slingbox_server.exe from outside of your network. When configured with the proper windows location that works!

And to ensure slingbox loads up if my pc crashes, I will continue to have a shortcut to slingbox_server.exe n my WIndows StartUp folder.
